### PR TITLE
assisted-service: create day2 master ZTP jobs

### PR DESCRIPTION
**Pull request by mkowalski**
Contributes-to: [MGMT-8578](https://issues.redhat.com//browse/MGMT-8578)
Contributes-to: [MGMT-12143](https://issues.redhat.com//browse/MGMT-12143)

### CODE REVIEW

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml [hunk 1]
```diff
@@ -489,6 +503,21 @@ tests:
...
+    dependencies:
+      ASSISTED_OPENSHIFT_INSTALL_RELEASE_IMAGE: release:latest
+      OPENSHIFT_INSTALL_RELEASE_IMAGE: release:latest
+    env:
+      ASSISTED_ZTP_CONFIG: |
+        SPOKE_CONTROLPLANE_AGENTS=1
+        NUMBER_OF_DAY2_HOSTS=1
+        DAY2_MASTERS=True
+      NUM_EXTRA_WORKERS: "2"
+      PACKET_PLAN: m3.large.x86
```
Comment (adriengentil) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:519]: how many VM does the job spawn? Try with the default machine I think

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml [hunk 2]
```diff
@@ -476,6 +476,20 @@ tests:
...
+  steps:
+    cluster_profile: packet-assisted
+    dependencies:
+      ASSISTED_OPENSHIFT_INSTALL_RELEASE_IMAGE: release:latest
+      OPENSHIFT_INSTALL_RELEASE_IMAGE: release:latest
+    env:
+      ASSISTED_ZTP_CONFIG: |
+        SPOKE_CONTROLPLANE_AGENTS=3
+        NUMBER_OF_DAY2_HOSTS=1
+        DAY2_MASTERS=True
```
Comment (omertuc) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:490]: Where is this used?

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml (hunk 1 above)
Comment (mkowalski) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:519]: I don't understand your question, it's defined just in the line you commented - `NUM_EXTRA_WORKERS: "2"`

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml (hunk 2 above)
Comment (mkowalski) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:490]: In https://github.com/openshift/assisted-service/pull/4437

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml (hunk 1 above)
Comment (adriengentil) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:519]: You set `PACKET_PLAN` to `m3.large` which is a big and more expensive machine, is there a specific reason? because I see the other job runs more VMs and you did not set `PACKET_PLAN` for it

Comment (mkowalski) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:519]: This job is a dumb copy of `e2e-ai-operator-ztp-sno-day2-workers`. Whatever is set there, I set here. Unless we want to discuss reconfiguration of the other job too, I don't see a reason why one should be different from the other

Comment (mkowalski) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:519]: To be clear

* `e2e-ai-operator-ztp-compact-day2-masters` is blind copy of `e2e-ai-operator-ztp-compact-day2-workers`
* `e2e-ai-operator-ztp-sno-day2-masters` is blind copy of `e2e-ai-operator-ztp-sno-day2-workers`

Comment (omertuc) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:519]: The job spawns 7 VMs - 5 for the hub (3 + 2 dev-scripts default), 2 for spoke (NUM_EXTRA_WORKERS)

Comment (mkowalski) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:519]: I'm not sure the discussion about those numbers really should be happening here. We have current jobs that take **some cluster** and add **some workers** to it. The new job will take **the same cluster** and **instead** of adding some workers it will add some masters.

We are not adding more nodes right now. It's just that previously we were adding workers, now we are adding masters.

>How does it even work? Are you allowed to add a master to a cluster with already 3 masters?

Yes, that's a new feature that we implement as MGMT-8578

Comment (mkowalski) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:519]: Number-wise everything is good. In the compact cluster we have

* `SPOKE_CONTROLPLANE_AGENTS=3`
* `NUMBER_OF_DAY2_HOSTS=1`
* `NUM_EXTRA_WORKERS: "4"`

In the SNO we have

* `SPOKE_CONTROLPLANE_AGENTS=1`
* `NUMBER_OF_DAY2_HOSTS=1`
* `NUM_EXTRA_WORKERS: "2"`

All the numbers add up correctly. We have enough VMs to address all the needs

Comment (omertuc) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:519]: yeah sorry I was mixing up the compact and SNO jobs

Comment (omertuc) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:519]: >How does it even work? Are you allowed to add a master to a cluster with already 3 masters?

However, this question was directed at OCP - not at Assisted

Comment (mkowalski) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:519]: >However, this question was directed at OCP - not at Assisted

Long story short, yes - in the process of replacing a master node in the OCP cluster there is a period of time when you have a cluster with 4 masters. It only an intermediate step that's supposed to be only for a short while, but nevertheless there will be a notion of cluster with 4 masters

Comment (omertuc) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:519]: OK. I'll be surprised if it works with a SNO - guess we'll see

Comment (mkowalski) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:519]: You are right, probably "replacing master" in case of SNO is a nonsense. Let me remove this test because I don't see any benefit

Comment (omertuc) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml]: It's not nonsense, it's definitely not officially supported though. SNO+Workers is supported since 4.11 and having temporarily 2 SNOs while the older one is being removed is not out of the question, even though it's not something we should be wasting our time on until it's supported

tl;dr not entirely nonsense but unsupported
